### DISCUSSION TITLE
com.microsoft.identity.client.exception.MsalClientException for Intune Company Portal

### DISCUSSION THREAD

**Discussion by arpusspzoo** [AzureAD/microsoft-authentication-library-for-android#1686] [Q&A] [answered]
Hello. 

We create an android application (MSAL 2.+) with AAD integration and an intune company portal.

I've generated a URI Redirect and registered application on AAD.
I want to use Intune Company Portal as broker.

- When broker_redirect_uri_registered=false -> application receives an ID token correctly
- When broker_redirect_uri_registered=true -> I get an exception (com.microsoft.identity.client.exception.MsalClientException: The redirect URI it the configuration file doesn’t match with the one generated with package name and signature hash. Please verify the uri in the config file and your app registration in Azure portal.)

`{
  "client_id" : "ABC",
  "authorization_user_agent" : "DEFAULT",
  "redirect_uri" : "msauth://xyz/def",
  "account_mode" : "SINGLE",
  "broker_redirect_uri_registered": true,
  "authorities" : [
    {
      "type": "AAD",
      "audience": {
        "type": "AzureADMultipleOrgs",
        "tenant_id": "organizations"
      }
    }
  ]
}`

What am I doing wrong?  

Sorry if this place is not good for questions like that:)

thanks

**Comment by shahzaibj** [comment, marked as answer]
Hi @arpusspzoo When `broker_redirect_uri_registered` is set to `true` then you need to make the redirect uri match the following format:

` "redirect_uri" : "msauth://<YOUR_PACKAGE_NAME>/<YOUR_BASE64_URL_ENCODED_PACKAGE_SIGNATURE>",`

You can follow the instructions in our quickstart and that would guide you towards creating the redirect uri correctly: https://learn.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app#add-a-redirect-uri
